lib/fog/vsphere/requests/compute/cloudinit_to_customspec.rb

```diff
@@ -0,0 +1,28 @@
+module Fog
+  module Compute
+    class Vsphere
+      class Real
+        def cloudinit_to_customspec(user_data)
+          raise ArgumentError, "user_data cant be nil" if user_data.nil?
+          custom_spec = { 'customization_spec' => Hash.new }
+          user_data = YAML.load(user_data)
+          custom_spec['hostname']                    =  user_data['hostname'] if user_data.key?('hostname')
+          custom_spec['ipsettings']                  =  { 'ip' => user_data['ip'] } if user_data.key?('ip')
+          custom_spec['ipsettings']['subnetMask']    =  user_data['netmask'] if user_data.key?('netmask')
+          custom_spec['ipsettings']['dnsServerList'] =  user_data['dns'] if user_data.key?('dns')
+          custom_spec['domain']                      =  user_data['domain'] if user_data.key?('domain')
+          custom_spec['dnsSuffixList']               =  user_data['domain'] if user_data.key?('domain')
+          custom_spec['time_zone']                   =  user_data['timezone'] if user_data.key?('timezone')
+          custom_spec           
+        end
+      end
+
+      class Mock
+        def cloudinit_to_customspec(user_data)
+          raise ArgumentError, "user_data cant be nil" if user_data.nil?
+          true
```

Comment:
We try to make the mocks reflect the real operations as closely as possible.  Granted, the codebase is pretty scattershot on this, but it is the goal we aspire to.  Is there a reason you couldn't return the same hash as the real implementation?  It doesn't look like you're making a 3rd party API call, so I'd just even delegate the mock to the real in this case.